documentation/development.md

```diff
@@ -0,0 +1,389 @@
+            break;
+        ...
+    }
+}
+```
+
+Lastly, we render the interface:
+
+```js
+function renderOperator(storageHandler: StorageHandler) {
+    const layout = storageHandler.loadCurrentLayoutOrDefault();
+
+    !isMobile
+        ? root.render(
+              <Operator
+                  remoteStreams={allRemoteStreams}
+                  layout={layout}
+                  storageHandler={storageHandler}
+              />,
+          )
+        : root.render(
+              <MobileOperator
+                  remoteStreams={allRemoteStreams}
+                  storageHandler={storageHandler}
+              />,
+          );
+}
+```
+
+Adding new capabilities to the web interface may involve creating a function provider. Let's take `HomeTheRobotFunctionProvider` for example. We instantiate it like so:
+
+```js
+export var homeTheRobotFunctionProvider: HomeTheRobotFunctionProvider = new HomeTheRobotFunctionProvider();
+```
+
+You will need to import it as well, but you might see "unresolvable module" errors until we create the `HomeTheRobotFunctionProvider.tsx` file in the [`function_providers/`](#function_providers) folder.
+
+### `function_providers/`
+
+The `function_providers/` folder contains all of the interface's "function providers", which are classes that literally provide functionality as anonmous functions, so that a UI component can map user actions to function calls. We'll create `HomeTheRobotFunctionProvider.tsx` with the following code:
+
+```js
+import { FunctionProvider } from "./FunctionProvider";
+import { HomeTheRobotFunction } from "../layout_components/HomeTheRobot";
+
+export class HomeTheRobotFunctionProvider extends FunctionProvider {
+
+    constructor() {
+        super();
+        this.provideFunctions = this.provideFunctions.bind(this);
+    }
+
+    public provideFunctions(homeTheRobotFunction: HomeTheRobotFunction) {
+        switch (homeTheRobotFunction) {
+            case HomeTheRobotFunction.Home:
+                return () => {
+                    FunctionProvider.remoteRobot?.homeTheRobot();
+                };
+        }
+    }
+}
+
+```
+
+Notice that `HomeTheRobotFunctionProvider` is a subclass of `FunctionProvider`. It is an abstract class and your function providers should extend it as well.
+
+Notice that we use `remoteRobot` to call into `homeTheRobot()`, which we defined in [`shared/remoterobot.tsx`](#remoterobottsx).
+
+Lastly, notice that we import `HomeTheRobotFunction` from `layout_components/`. It is an interface that defines all the functions that the UI component will need. For this example, there's only one function: homing. We will define this interface in the next section.
+
+### Components
+
+There are three component folders:
+
+- `layout_components/`: for components that will be displayed as part of the interface's layout. These are typically top-level components.
+- `static_components/`: TODO - what is the difference between a static component and a basic component??
```

Comment:
@hello-vinitha Would you mind clearing up the difference between static and basic components for me?

Reply:
@hello-vinitha, quick ping, in case you didn't see this